recipe-server/normandy/recipes/api/views.py

```diff
@@ -129,6 +131,26 @@ def disable(self, request, pk=None):
         recipe.save()
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @detail_route(methods=['POST'])
+    def request_approval(self, request, pk=None):
+        recipe = self.get_object()
+
+        if recipe.is_approved:
+            return Response({'error': 'Recipe is already approved.'},
+                            status=status.HTTP_400_BAD_REQUEST)
+        elif not recipe.latest_revision:
+            return Response({'error': 'Recipe has no revisions to be approved.'},
+                            status=status.HTTP_400_BAD_REQUEST)
+        elif recipe.latest_revision.is_rejected:
+            return Response({'error': 'The latest revision has already been rejected. Please make '
+                                      'changes before requesting approval.'},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        revision = recipe.latest_revision
+        revision.approval_request = ApprovalRequest(revision=revision, user=request.user)
+        revision.save()
```

Comment:
This seems racey. Should we include the current revision id along with the request approval?